### ISSUE TITLE
Multilanguage GUI

### ISSUE THREAD

**Issue by bovirus** [tonikelope/megabasterd#397]
@tonikelope 

Can you help to improve the program making a multilanguage GUI?

If you enable this option I can translate it in Italian.

Thanks.

**Comment by tonikelope** [comment]
Hello. MegaBasterd already has support for adding languages. Here is a link to the template for adding new languages (translating from English). When you have it finished email it to me or create a pate on pastebin.com and I will add it in the next update.

Thanks for your help.

https://pastebin.com/1wXDWZX9

**Comment by bovirus** [comment]
@tonikelope 

Thanks for the info.

What languages are currently included in the program?

I propsoe to maek this important info how to translate on home page in github repository.

I believe that you will find a lot of pepole that want to help you to create new languages.

**Comment by tonikelope** [comment]
Currently English and Spanish. Currently English and Spanish. But as I said, you can add as many as you want.

**Comment by bovirus** [comment]
@tonikelope 

Propsoal for home page on github

**_- Language available_**

English
Spanish

**_- Create  anew language_**

[url=https://github.com/tonikelope/megabasterd/issues/397]_How to create a ne language_[/url]

**Comment by bovirus** [comment]
@tonikelope 

Is there an option to test new language in place before to send you t include in next release?

**Comment by enagonius** [comment]
I think I will translate it to Portuguese!

**Comment by bovirus** [comment]
@tonikelope 

Translation for italian

https://mega.nz/file/bRRx2CqB#almQa2pCdDrbS0KpiMiD6NmaVu_LiyiKQWy4LSD_yDQ

**Comment by FabrieI** [comment]
@tonikelope 

New translate (Turkish)

https://github.com/FabrieI/megabasterd_translate

Thanks,

**Comment by bovirus** [comment]
@tonikelope

Update italian language

[MegaBasterd - italian.txt](https://github.com/tonikelope/megabasterd/files/8813775/MegaBasterd.-.italian.txt)

Please add italian and turkish language.

**Comment by FabrieI** [comment]
@bovirus @tonikelope 

Updated, turkish language (Some fixes,dialog)

https://github.com/FabrieI/megabasterd_translate

**Comment by bovirus** [comment]
@tonikelope 

New italian language file

[MegaBasterd - italian.txt](https://github.com/tonikelope/megabasterd/files/8837211/MegaBasterd.-.italian.txt)


@tonikelope 
@fabriel

Isues about GUI).
Strings not translated or other issues.

![image](https://user-images.githubusercontent.com/1262554/171989062-f149a170-d217-4ef2-9e48-b7471ac636a4.png)

String not translated and not availabel in language file
"CANCEL ALL DOWNLOADS" (eng)
"ANNULLA TUTTI I DOWNLOAD" (ita)

![image](https://user-images.githubusercontent.com/1262554/171989097-096e8448-e3d4-4863-bbf5-b33a297433e6.png)

String not translated and not availabel in language file
Save debug info to file" (eng)
"Salva info debug nel file" (ita)

![image](https://user-images.githubusercontent.com/1262554/171989150-6cdb04ac-aed5-4dcd-ad91-32e934eb58e4.png)

"Use this proxy list Format js [*]IP:PORT[@user_b64:password_b64)" (eng)
"Usa questo formato per l'elenco proxy  [*]IP:PORT[@user_b64:password_b64)" (ita)

@tonikelope 

Please update your master language with these three new strings (useful for new translators).

**Comment by FabrieI** [comment]
Thank you @bovirus 

Fixes and added news strings.

Updated turkish language (5 june 2022)

https://github.com/FabrieI/megabasterd_translate

**Comment by FabrieI** [comment]
Thanks again @bovirus 

Added news strings.

https://github.com/FabrieI/megabasterd_translate

**Comment by bovirus** [comment]
@tonikelope 

Could you please rebuild a new version with latest version of language strings?
Thanks.

**Comment by linkea131** [comment]
@bovirus @tonikelope

Updated, Chinese language (garbled And Some fixes,dialog)
https://mega.nz/file/fTR3UDSK#X4UfEvzoQbNgEL7dwvHOWRAVIBaY3xl8T_ecZUXV8Kc

**Comment by linkea131** [comment]
@bovirus @tonikelope

Please help to solve the Chinese garbled problem Thank you

https://ru.xdnsw.xyz/garbled.png

Text encoding has been modified to UTF-8 and GB2312 encoding
https://ru.xdnsw.xyz/GB2312/MegaBasterd - Chinese.txt
https://ru.xdnsw.xyz/UTF-8/MegaBasterd - Chinese.txt

**Comment by linkea131** [comment]
@bovirus @tonikelope

Please help to solve the Chinese garbled problem Thank you

https://ru.xdnsw.xyz/garbled.png

Modify the text encoding to GB2312
https://ru.xdnsw.xyz/GB2312/MegaBasterd - Chinese.txt

**Comment by linkea131** [comment]
@bovirus @tonikelope

Please help to solve the Chinese garbled problem Thank you

![garbled](https://user-images.githubusercontent.com/50718406/174442300-1730e1c7-074d-414d-b511-3deb585e97ea.png)

Modify the text encoding to GB2312
https://ru.xdnsw.xyz/GB2312/MegaBasterd - Chinese.txt

**Comment by linkea131** [comment]
@bovirus @tonikelope

Updated, Chinese language (Some fixes,dialog 18 june 2022)

https://github.com/linkea131/megabasterd_translate

**Comment by linkea131** [comment]
#https://github.com/linkea131/megabasterd_translate/commit/c4623315b8a24798bb017240edc553e09c84ed31

**Comment by bovirus** [comment]
@tonikelope 

Thanks for new version.

If we will find other strings to revise we will open a new issue.

**Comment by NieckLikesCode** [comment]
German translation:
[MegaBasterd German Translation.txt](https://github.com/tonikelope/megabasterd/files/8978530/MegaBasterd.German.Translation.txt)

**Comment by bovirus** [comment]
@tonikelope 

What do you think to modify source code to use for example external .json files for languages?

**Comment by bovirus** [comment]
@tonikelope 
@linkea131 
@NieckLikesCode 

Other untranslated strings to add to add to the language file

![image](https://user-images.githubusercontent.com/1262554/176128528-35f435b6-f713-4f7d-a481-df0d5cb572c7.png)

![image](https://user-images.githubusercontent.com/1262554/176128576-16fbb2c5-9713-4cd6-9787-2b017754ac82.png)

![image](https://user-images.githubusercontent.com/1262554/176128645-02cef15c-c013-40c3-a5bb-595233525ffe.png)

        _addTranslation("Link Grabber", "");
        _addTranslation("Select DLC Container", "");
        _addTranslation("No MEGA accounts available (Go to Settings > Accounts)", "");
        _addTranslation("German", "");

The string German is for new German language.

@tonikelope 

There is abug regarding show "MC reverse: OFF" string always in english aslo if is included in language file

![image](https://user-images.githubusercontent.com/1262554/176128835-8c064095-f89a-4245-9406-ed21caae81d6.png)

**Comment by linkea131** [comment]
@bovirus @tonikelope

Updated, Chinese language (Some fixes,dialog July 1 2022)

https://github.com/linkea131/megabasterd_translate

**Comment by bovirus** [comment]
@linkea131 

Updatd chinese language following rules of internal files

[MegaBasterd - Language - Chinese.txt](https://github.com/tonikelope/megabasterd/files/9027254/MegaBasterd.-.Language.-.Chinese.txt)

@NieckLikesCode

Update German language following rules of internal file.
If you can pelase update langaueg adding the new strings found (see below)

[MegaBasterd - Language - German.txt]
(https://github.com/tonikelope/megabasterd/files/9027265/MegaBasterd.-.Language.-.German.txt)

**Comment by bovirus** [comment]
@linkea131 
@NieckLikesCode 

Here you cam find new .jar developed with your language update.
Downlaod and replace the file.

https://mega.nz/file/3VBjxbRT#b24m8C8PrEAcTO-ydb1iuDjUFoTyzSMd_84DX_z8G4w

@NieckLikesCode 

I eneabled the German language setup from drodown list but show it in English.
I'm working on it.
I hope that @tonikelope fix it.

**Comment by bovirus** [comment]
@NieckLikesCode 

Toa void long post is better not copy all rpevious post

It seems that your file has a syntax problem 

_addTranslation("Link Grabber", "");_addTranslation("Select DLC container", "");

should be

_addTranslation("Link Grabber", "");
_addTranslation("Select DLC container", "");

and three empty strings

_addTranslation("Link Grabber", "");
_addTranslation("Select DLC container", "");
_addTranslation("No MEGA accounts available (Go to Settings > Accounts)", "");

Here the file compatible with the syntax of script

[MegaBasterd - Language - German.txt](https://github.com/tonikelope/megabasterd/files/9029972/MegaBasterd.-.Language.-.German.txt)

pelase sue it and fillin three empty translated strings.

**Comment by linkea131** [comment]
> 

good bad syntax removed fix 3 Jul 2022

**Comment by linkea131** [comment]
@bovirus @tonikelope

Updated, Chinese language (bad syntax removed fix 3 Jul 2022 )

https://github.com/linkea131/megabasterd_translate

**Comment by bovirus** [comment]
@linkea131 - Chinese language
@NieckLikesCode - German language
@FabrieI - Turkish language
Spanish language - ?????

**English language (master)**
[MegaBasterd - Language - English.txt](https://github.com/tonikelope/megabasterd/files/9078775/MegaBasterd.-.Language.-.English.txt)

**Chinese language**
[MegaBasterd - Language - Chinese.txt](https://github.com/tonikelope/megabasterd/files/9078778/MegaBasterd.-.Language.-.Chinese.txt)

**_German language_**
[MegaBasterd - Language - German.txt](https://github.com/tonikelope/megabasterd/files/9078780/MegaBasterd.-.Language.-.German.txt)

**_Italian language_**
[MegaBasterd - Language - Italian.txt](https://github.com/tonikelope/megabasterd/files/9078781/MegaBasterd.-.Language.-.Italian.txt)

**_Spanish language_**
[MegaBasterd - Language - Spanish.txt](https://github.com/tonikelope/megabasterd/files/9078784/MegaBasterd.-.Language.-.Spanish.txt)

**_Turkish language_**
[MegaBasterd - Language - Turkish.txt](https://github.com/tonikelope/megabasterd/files/9078788/MegaBasterd.-.Language.-.Turkish.txt)

**Comment by bovirus** [comment]
New jar beta jwith current language status (German language enabled)

https://mega.nz/file/rJ4BhYRZ#1FBq_d0EDS6tG0sW2watijS-QjlJl1Jk435oMbyzrpI

**Comment by linkea131** [comment]
@bovirus @tonikelope

Updated, Chinese language (dialog July 11, 2022)

https://github.com/linkea131/megabasterd_translate
![1](https://user-images.githubusercontent.com/50718406/178282114-9dd22648-d10d-4c35-aa4d-4ff572ea8ba1.png)
![2](https://user-images.githubusercontent.com/50718406/178282134-b4432016-4da4-48df-95b7-08dfa06e8573.png)
![3](https://user-images.githubusercontent.com/50718406/178282139-53426af6-4062-4efc-ba59-275dd46e4f38.png)
![4](https://user-images.githubusercontent.com/50718406/178282148-faa73bab-80b2-4b97-8881-4bf0baa8e950.png)

**Comment by bovirus** [comment]
@linkea131 

Why didn't translate

        _addTranslation("English", "");
        _addTranslation("Spanish", "");
        _addTranslation("Italian", "");
        _addTranslation("Turkish", "");
        _addTranslation("Chinese", "");
        _addTranslation("German", "");

Thanks. KR.

**Comment by linkea131** [comment]
> @linkea131
> 
> ```
>     _addTranslation("MEGA API KEY)", "");
> ```
> 
> should be
> 
> ```
>     _addTranslation("MEGA API KEY:", "");
> ```
> 
> and seems untranslated.
> 
> These strings seems untranslated
> 
> ```
>     _addTranslation("MEGA API KEY ERROR", "");
>     _addTranslation("An older version of MegaBasterd has been detected (", "");
>     _addTranslation(").\nDo you want to import all current settings and transfers from the previous version?\nWARNING: INCOMPATIBILITIES MAY EXIST BETWEEN VERSIONS.", "");
>     _addTranslation("All your transferences have finished", "");
>     _addTranslation("Email", "");
> 
>     _addTranslation("API-KEY", "");
> 
>     _addTranslation("MC reverse mode: OFF", "");
>     _addTranslation("Restore MegaBasterd window", "");
>     _addTranslation("Exit from MegaBasterd", "");
>     _addTranslation("English", "");
>     _addTranslation("Spanish", "");
>     _addTranslation("Italian", "");
>     _addTranslation("Turkish", "");
>     _addTranslation("Chinese", "");
>     _addTranslation("German", "");
> ```
> 
> Here your file with _addTranslation("MEGA API KEY:", ""); fix.
> 
> [MegaBasterd - Language - Chinese.txt](https://github.com/tonikelope/megabasterd/files/9084920/MegaBasterd.-.Language.-.Chinese.txt)
> 
> Thanks.

The new text has been translated. I think people who do not understand English can understand the parts that have not been translated, so the parts that can be understood will not be translated.

**Comment by bovirus** [comment]
@linkea131 

Yes but if you don't want to translate you have to option. Ex:

    _addTranslation("English", "");

or remove it or fillin the scecond part with english

    _addTranslation("English", "English");

othewrise if you elave

    _addTranslation("English", "");

instead "English" in your language show empty string,

**Comment by FabrieI** [comment]
Thanks friend.

@bovirus 
@tonikelope 

Fixes: String line & Editor: Notepad++

21
Original
_addTranslation("(FROZEN) Waiting to start (", " ");

Fixed
_addTranslation("(FROZEN) Waiting to start **"),** " ");


40
Original
_addTranslation("You have the latest version ;)", "");

Fixed
_addTranslation("You have the latest version **"),** " ");



250
Original
_addTranslation("Streaming server: ON (port ", "");

Fixed
_addTranslation("Streaming server: ON (port **"),** " ");


251
Original
_addTranslation("MC reverse mode: ON (port ", "");

Fixed
_addTranslation("MC reverse mode: ON (port **"),** " ");


252
Original
_addTranslation("Streaming server: OFF (port ", "");

Fixed
_addTranslation("Streaming server: OFF (port **")**, " ");


253
Original
_addTranslation("MC reverse mode: OFF (port ", "");

Fixed
_addTranslation("MC reverse mode: OFF (port **"),** " ");


271
Original
_addTranslation("No MEGA accounts available (Go to Settings > Accounts)", "");

Fixed
_addTranslation("No MEGA accounts available (Go to Settings > Accounts) **)",** " ");

272
Original
_addTranslation("MEGA API KEY", "");

Fixed
_addTranslation("MEGA API KEY **"),** " ");    


273
Original
_addTranslation("MEGA API KEY ERROR", "");   
 
Fixed
_addTranslation("MEGA API KEY ERROR **"),** " ");         


https://github.com/FabrieI/megabasterd_translate

**Comment by bovirus** [comment]
@FabrieI 

Please test new jar file with your translation

https://mega.nz/file/bJomiKBR#g09zEYOpMDibh-ErWRJlwLZQ8eM3ZW1s8i8bmu-81t8

**Comment by linkea131** [comment]
> @linkea131
> 
> **Empty strings**
> 
> Yes but if you don't want to translate you have to option. Ex:
> 
> ```
> _addTranslation("English", "");
> ```
> 
> or remove it or fill-in the seccond part with same english text (suggested option)
> 
> ```
> _addTranslation("English", "English");
> ```
> 
> othewrise if you leave
> 
> ```
> _addTranslation("English", "");
> ```
> 
> instead "English" in your language show empty string,
> 
> **_String errors_**
> 
> The language file has some errros (double quotes)
> 
> wrong
> 
> _addTranslation("FILE WITH SAME NAME AND SIZE ALREADY EXISTS", "已存在相同名称大小的文件""); _addTranslation("Port:", ""端口:"");
> 
> right
> 
> _addTranslation("FILE WITH SAME NAME AND SIZE ALREADY EXISTS", "已存在相同名称大小的文件"); _addTranslation("Port:", "端口:");
> 
> _addTranslation("MEGA API KEY)", "");
> 
> should be
> 
> _addTranslation("MEGA API KEY:", "");
> 
> **_Empty strings_**
> 
> These strings are empty
> 
> _addTranslation("MEGA API KEY)", ""); _addTranslation("Email", ""); _addTranslation("API-KEY", ""); _addTranslation("English", ""); _addTranslation("Spanish", ""); _addTranslation("Italian", ""); _addTranslation("Turkish", ""); _addTranslation("Chinese", ""); _addTranslation("German", "");
> 
> should be fill-in with english words
> 
> _addTranslation("MEGA API KEY:", "MEGA API KEY:"); _addTranslation("Email", "Email"); _addTranslation("API-KEY", "API-KEY"); _addTranslation("English", "English"); _addTranslation("Spanish", "Spanish"); _addTranslation("Italian", "Italian"); _addTranslation("Turkish", "Turkish"); _addTranslation("Chinese", "Chinese"); _addTranslation("German", "German");
> 
> Thanks.

@bovirus @tonikelope

Updated, fixe string Chinese language (dialog July 21, 2022)

https://github.com/linkea131/megabasterd_translate

**Comment by bovirus** [comment]
@linkea131 

Please check your github repository.
In your past version there were some errros that I tried to fix making a PR but you din't merge it

https://github.com/linkea131/Megabasterd-GUI-Translate/pulls

I fixess the syntax error (typicalli a " missing) in this file.

[MegaBasterd - Language - Chinese.txt](https://github.com/tonikelope/megabasterd/files/9160011/MegaBasterd.-.Language.-.Chinese.txt)

Please use it to update your repository.

Thanks.

**Comment by bovirus** [comment]
New jar file with update about chinese language

https://mega.nz/file/HZRTAQZD#WsAj3n-xAc7uuyEglp-ER3B2GvewVixTC3Xy1HDcZUw

**Comment by Roschach96** [comment]
@tonikelope

Hi, I just made the hungarian language translation, please add it in one of the next updates.
(I added 3 extra lines: Italian, Chinese and Turkish languages because these were missing from the file.)

[Hungarian.txt](https://github.com/tonikelope/megabasterd/files/9174662/Hungarian.txt)

**Comment by bovirus** [comment]
@Roschach96 

here are some strings missings
Could you pelase complete the translation file? Thanks.

[MegaBasterd - Language - Hungarian.txt](https://github.com/tonikelope/megabasterd/files/9175458/MegaBasterd.-.Language.-.Hungarian.txt)

Please use my file and edit it.
Then when you wills end back the file I will create a new .jar build for test.

**Comment by bovirus** [comment]
- Chinese language
[MegaBasterd - Language - Chinese.txt](https://github.com/tonikelope/megabasterd/files/9175469/MegaBasterd.-.Language.-.Chinese.txt)

- German language
[MegaBasterd - Language - German.txt](https://github.com/tonikelope/megabasterd/files/9175470/MegaBasterd.-.Language.-.German.txt)

- Spanish language
[MegaBasterd - Language - Spanish.txt](https://github.com/tonikelope/megabasterd/files/9175471/MegaBasterd.-.Language.-.Spanish.txt)

- Turkish
[MegaBasterd - Language - Turkish.txt](https://github.com/tonikelope/megabasterd/files/9175472/MegaBasterd.-.Language.-.Turkish.txt)

**Comment by bovirus** [comment]
new jat with current latest chanegs (add hungarian language).

https://mega.nz/file/XMYg0RQJ#XQz0gkrxe4sTeTpBTMUUgA75R9CTYzMP-WG3OVnsJJQ

**Comment by Roschach96** [comment]
@bovirus

I added the extra lines that you mentioned above.

[MegaBasterd.-.Language.-.Hungarian.txt](https://github.com/tonikelope/megabasterd/files/9175585/MegaBasterd.-.Language.-.Hungarian.txt)

**Comment by bovirus** [comment]
@Roschach96 

Thanks for the Hungarian language update.
This the jar built with your latest changes.

https://mega.nz/file/yZZx2C5Y#ngq8oS2gERzu-4w9THJ5ahTOl4m91yTPRtqq1okXlXU

**Comment by Roschach96** [comment]
> @Roschach96
> 
> Thanks for the Hungarian language update. This the jar built with your latest changes.
> 
> https://mega.nz/file/yZZx2C5Y#ngq8oS2gERzu-4w9THJ5ahTOl4m91yTPRtqq1okXlXU

I had to correct some mistyped text, I send the corrected versio
[MegaBasterd.-.Language.-.Hungarian.txt](https://github.com/tonikelope/megabasterd/files/9216779/MegaBasterd.-.Language.-.Hungarian.txt)
n

**Comment by bovirus** [comment]
@Roschach96 

Thanks. 
New jar version

https://mega.nz/file/6FhQgK6S#i8fHS6pYKKiR-4g8N3SDGa12qjCs8rS4eqEs7p84f_E

**Comment by linkea131** [comment]
@bovirus @tonikelope

Updated, fixe string Chinese language (dialog August 19, 2022)

https://github.com/linkea131/megabasterd_translate

**Comment by imnotdev25** [comment]
Soon Adding Hindi & Gujarati support

**Comment by bovirus** [comment]
@linkea131 

New jar files with all your latest changes.

https://mega.nz/file/eMo3kaoI#3klVBjhi9CqGS9XcaSz5DqlbiHcsc6b4XM1D_RLtbnM

Is it right that these strings remained in english?

        _addTranslation("English", "English");
        _addTranslation("Spanish", "Spanish");
        _addTranslation("Italian", "Italian");
        _addTranslation("Turkish", "Turkish");
        _addTranslation("Chinese", "Chinese");
        _addTranslation("German", "German");

**Comment by sumanbyte** [comment]
Hello Sir, Can I translate this application into nepali language ? How can i do so? I want to contribute to this project by translating it to Nepali Language.

**Comment by linkea131** [comment]
> @linkea131
> 
> New jar files with all your latest changes.
> 
> https://mega.nz/file/eMo3kaoI#3klVBjhi9CqGS9XcaSz5DqlbiHcsc6b4XM1D_RLtbnM
> 
> Is it right that these strings remained in english?
> 
> ```
>     _addTranslation("English", "English");
>     _addTranslation("Spanish", "Spanish");
>     _addTranslation("Italian", "Italian");
>     _addTranslation("Turkish", "Turkish");
>     _addTranslation("Chinese", "Chinese");
>     _addTranslation("German", "German");
> ```

YEAH

**Comment by rattybox** [comment]
@tonikelope
Vietnamese language pack
https://pastebin.com/xjaj28xV
Thanks,

**Comment by Sneak71** [comment]
> Hello. MegaBasterd already has support for adding languages. Here is a link to the template for adding new languages (translating from English). When you have it finished email it to me or create a pate on pastebin.com and I will add it in the next update. Please RESPECT capitalization in translations and be careful not to modify any characters of the original English messages on the left side.
> 
> Thanks for your help.
> 
> https://pastebin.com/1wXDWZX9
> 
> `_addTranslation("FILE WITH SAME NAME AND SIZE ALREADY EXISTS", "ESISTE GIÀ UN FILE CON LO STESSO NOME E DIMENSIONE");`

Hey Tonikelope! Sup mate!? If you want i can help ya with the French translation! Just lemme know if you are still in need. I'm able to translate whatever you want every days (afternoon). Sadly my computer have an issue so i'll just translate every sentences you can send by mail or there on GitHub. Keep the good job up bro!
 See ya later, take care bro! ^_*
Best & kind regards.

**Comment by bovirus** [comment]
@Sneak71 

If you post the language I can make a temporary build to test it.

**Comment by tonikelope** [comment]
Hi folks. I don't have time to go through all the answers here. **I NEED HELP to gather all the translations into one file that I can add to the source code for 7.57.** THANK YOU.

MERRY CHRISTMAS AND A PEACEFUL 2023

**Comment by tonikelope** [comment]
https://github.com/tonikelope/megabasterd/issues/498#issuecomment-1374750765

**Comment by rktfier** [comment]
[Language_Turkish.txt](https://github.com/tonikelope/megabasterd/files/11243233/Language_Turkish.txt)
Here is my updated translations for Turkish. The old group seems to do MTL, so most of their translations are not great, and there are a lot of mistakes (if the group wants to contact me about it, please do so). If they want to continue translations, pleas let me know, or else i can try and take the translations, basically redoing nearly half of them. Still, even if you're doing MTL, at the very least try to use good ones like DeepL...

**Comment by bovirus** [comment]
@tonikelope 

Please modify source code to have separate file for each language.

Could be easier to mantain it for translators.

**Comment by bovirus** [comment]
@rktfier

Please check Turkish language file on first post.
I got your language and sorted it alphabetically.
I compared your fiel with master English and add strings missing.
Please complete the new strings (2) in Turkish..

**Comment by rktfier** [comment]
[Turkish.txt](https://github.com/tonikelope/megabasterd/files/11304974/Turkish.txt)

Translated lines 39 and 57, fixed a small (possible) error in line 237.

**Comment by bovirus** [comment]
@Roschach96

Could you please update Hungarian translation (get it from first post)?

@tonikelope 

Could you please update Spanish translation (get it from first post)?

**Comment by bovirus** [comment]
@rktfier 

Found a new string.
Please update Turkish language (get it from first post).

**Comment by rktfier** [comment]
[Turkish.txt](https://github.com/tonikelope/megabasterd/files/11311213/Turkish.txt)

Translated line 58.

**Comment by bovirus** [comment]
@rktfier 

Thanks.
Update first post with Turkish language file and new build of jar file.
Please download .jar file, replace current jar and test it.

**Comment by rktfier** [comment]
[Turkish.txt](https://github.com/tonikelope/megabasterd/files/11315615/Turkish.txt)

Still a lot more to do, 15 lines have been changed, if possible i'd like to get a new build with this new translations so i don't have to come back to the same lines over and over again. The old TL also has their name/brand in a couple of places (examples for emails, and the about tab so far) so let me know if you'd like them to be removed or not. I'll not be rebranding or putting my name in any of the translations unless it's required. 




> @linkea131
> 
> New jar files with all your latest changes.
> 
> https://mega.nz/file/eMo3kaoI#3klVBjhi9CqGS9XcaSz5DqlbiHcsc6b4XM1D_RLtbnM
> 
> Is it right that these strings remained in english?
> 
> ```
>     _addTranslation("English", "English");
>     _addTranslation("Spanish", "Spanish");
>     _addTranslation("Italian", "Italian");
>     _addTranslation("Turkish", "Turkish");
>     _addTranslation("Chinese", "Chinese");
>     _addTranslation("German", "German");
> ```

Also based on this, should these strings stay in English? Currently the Turkish TL changes them to Turkish, which might be hard for users to change after accidentally clicking one of the translations. To be honest it could be better if we changed them to their native language across all translations, for example : ("Chinese", "中国人");

**Comment by bovirus** [comment]
@rktfier 

Please take care that to go Settings and select language you should knwo the language.

The idea could be translate in native language with abbreviation of the language. Example in Italian

Cinese (CH)
Tedesco (DE)
Italian (IT)

etc.

**Comment by bovirus** [comment]
@rktfier

Did you check new jar with full Turkish translation?

@linkea131
Could you please update Chinese translation (get it from first post)?

@NieckLikesCode
Could you please update German translation (get it from first post)?

@Roschach96
Could you please update Hungarian translation (get it from first post)?

@tonikelope
Could you please update Spanish translation (get it from first post)?

@rattybox
Could you please update Vietnamese translation (get it from first post)?

**Comment by rktfier** [comment]
I have had a busy week so i haven't been able to check it out until today. The new jar seems to refuse to run for me, i can see the splash screen but it quickly disappears after that.

New translations are here: 
[Turkish.txt](https://github.com/tonikelope/megabasterd/files/11358842/Turkish.txt)

Removed old translator's branding from line 6, and line 170.

Lines 47, 108, 113, 120, 241, 262, 291 which are the language ones, have been changed.

For the other translators out there, here is my current lines :
```
47  中国人 (ZH)
108 Deutsch (DE)
113 Magyar (HU)
120 Italiano (IT)
241 Español (ES)
262 Türkçe (TR)
291 Tiếng Việt (VI)
```

**Comment by bovirus** [comment]
@rktfier

What version of Java are you using?
My java version is 1.8.0.371

**Comment by rktfier** [comment]
I'm currently using 1.8.0_371, from the Oracle's website, i can try switching to Adoptium though.

**Comment by bovirus** [comment]
@rktfier

There was a syntax error in your latest file about position of "
Fixed.
Updated on first post Turkish language and jar file

![image](https://user-images.githubusercontent.com/1262554/235297784-1b51233b-db9a-4d25-adaa-4856066931a8.png)

Turkish GUI

**Comment by rktfier** [comment]
Seems like Chinese characters don't want to be seen:
![image](https://user-images.githubusercontent.com/75695425/235298736-04a2cc0f-4d70-4f6f-b225-e4e4f6d1f64c.png)

In the meantime, is this line hard coded? If we indeed translated all lines, then it should be in Turkish as well:
![image](https://user-images.githubusercontent.com/75695425/235298788-90a8beee-ee47-4254-b514-57d4ffdfe047.png)

**Comment by bovirus** [comment]
@rktfier

Pease get archive extract Tirkish language file.
Update chinese language name and translate new string just added (your report).

**Comment by rktfier** [comment]
Updated here :
[Turkish.txt](https://github.com/tonikelope/megabasterd/files/11359065/Turkish.txt)

Also have you considered making another repository so we can work and send our changes as pull requests? I'd say it would be a great idea including actions and checks so issues can be checked automatically.

**Comment by Roschach96** [comment]
Updated here:
[Hungarian.txt](https://github.com/tonikelope/megabasterd/files/11359201/Hungarian.txt)

**Comment by bovirus** [comment]
@rktfier 
@Roschach96 

Thanks for language updates.
I updated in first post the language archive file with your language updates and I created a new jar build.
Please test new JAR file.

**Comment by Roschach96** [comment]
Hungarian tested. Now everything is Hungarian for me.

**Comment by bovirus** [comment]
@rktfier 
@Roschach96

Update jar.
Now in the title and in tray icon show (mod dd.mm.yyyy) where dd.mm.yyyy is the compilation date,

**Comment by bovirus** [comment]
@linkea131
Could you please update Chinese translation (get it from archive in first post)?

@NieckLikesCode
Could you please update German translation (get it from archive in first post)?

@tonikelope
Could you please update Spanish translation (get it from archive in first post)?

@rattybox
Could you please update Vietnamese translation (get it from archive in first post)?

@rktfier
@Roschach96 

Could you pelase confirm that latest build(with build date on top left) is ok?

**Comment by Roschach96** [comment]
I can confirm that Hungarian translation is working fine with latest build(with build date on top left).

**Comment by bovirus** [comment]
@Roschach96 

Thanks.
Please check that I asked if you are intersted to Passy (other Github project) translation.

**Comment by rktfier** [comment]
Turkish TL also looks fine for the most part.

**Comment by bovirus** [comment]
@rktfier

"most of the part"?
Are there some part not right?

**Comment by rktfier** [comment]
Haven't you read me xomplaining about how more than half of this is MTL? (not even good MTL)

**Comment by bovirus** [comment]
@rktfier 

You have the option to change about the langauage what you want.

**Comment by rktfier** [comment]
Line 171 in the English translations might be wrong, there is no "Opening" a port, the process is called "Forwarding".

I checked the whole translation file until line 244, more than 30 lines have been changed here it is:
[Turkish.txt](https://github.com/tonikelope/megabasterd/files/11414388/Turkish.txt)

**Comment by bovirus** [comment]
@rktfier 
New jar build and langauge file archive available.
Please test new build.

**Comment by rktfier** [comment]
The strings here seems to not exist on the TL files.

![image](https://github.com/tonikelope/megabasterd/assets/75695425/aaf2d2f1-338c-45d6-92ad-ec8647c8db6e)

I have noticed a few more inconsistency errors in the translations, probably going to fix after these strings are possibly added.

**Comment by bovirus** [comment]
@rtkfier

It could require change to source code.

**Comment by rktfier** [comment]
Yeah that's what i initially thought. I also can't find the strings for the following, considering the translations is 100%  :
![image](https://github.com/tonikelope/megabasterd/assets/75695425/4a756ebf-ee08-4a5b-b334-5d08228934f3)

**Comment by bovirus** [comment]
@rktfier
Updated in first post languages (add new string).
Please complete string in Turkish language file.

**Comment by linkea131** [comment]
@bovirus yeah I can update the Chinese language translation, please provide me with the new language file

**Comment by rktfier** [comment]
Turkish file will be updated.

**Comment by KnightChaser** [comment]
@tonikelope 
@bovirus 

Translation for Korean

https://github.com/KnightChaser/megabastardTranslationKR

**Comment by bovirus** [comment]
@KnightChaser 

Please read carefully the first post of this thread.

You used for your translation old original version of megabasterd language file 

256 strings not ordered alphabetically

then new language file (english - 309 strings - ordered alphabetically) 

posted in first post.

I re-ordered your Korean language file (256 strings) alphabetically.

[Language_Korean.txt](https://github.com/tonikelope/megabasterd/files/11712943/Language_Korean.txt)

Please use it to compare with english to find the missing 53 string in yor language file.

Then posted Korean language file updated (should have 309 strings).

**Comment by KnightChaser** [comment]
@bovirus 

Apologies for the misunderstanding and for submitting the old version. 

However, the file `Language_Korean.txt` that you suggested doesn't seem to be alphabetically ordered, even though you claimed you reordered the list.

Anyway, I'll alphabetically order the list and fill in the missing 53 strings to make it complete right away.

**Comment by KnightChaser** [comment]
@bovirus 

Fixed to the new version. It now exactly has 309 strings now and is also alphabetically ordered. Please check the following repository again and 

https://github.com/KnightChaser/megabasterdTranslationKR

**Comment by bovirus** [comment]
@KnightChaser 

Thanks.
Please check that there were two errors in lines

34 - CancelANCEL -> CANCEL
84 - EXITxit -> Exit

Tomorrow I will build the new jar files.

**Comment by KnightChaser** [comment]
@bovirus 

Thanks for the fast feedback, Just fixed the suggested two typos that you mentioned.

https://github.com/KnightChaser/megabasterdTranslationKR/commit/fcff9240ea7ce67030fb6a74180c715b3b861a91

**Comment by bovirus** [comment]
@KnightChaser

Update language archive in first post with Korean language.

I added Korean translation for "Korean" language name.
Please check if it's ok.

Test the new jar in first post.

**Comment by KnightChaser** [comment]
@bovirus 

I looked at the Korean translation file again and fixed some miscellaneous mistakes(or errors that may be ignored) including the translation of the word `Korean`(`한국인`(_meaning Korean people_) → `한국어`(_meaning Korean language_)). Except for those, everything looks okay currently.

(▼ The latest Korean translation(fixed). Please update the language file.)

https://github.com/KnightChaser/megabasterdTranslationKR/commit/8ec107eda37f9dfef889ac8f4a45c67243d1e79a

**Comment by bovirus** [comment]
@KnightChaser 

Done. Update Korean language and jar file.
Please test it.

**Comment by KnightChaser** [comment]
@bovirus 

Is the JAR file download link `https://mega.nz/file/CR5TiRjJ#MPGIdHnbmJD4w58u8CpnNenzu0NsWtMkQFfcjS11EJ4` which is written on the first post of this thread, right? It seems to be inaccessible now.

https://mega.nz/file/CR5TiRjJ#MPGIdHnbmJD4w58u8CpnNenzu0NsWtMkQFfcjS11EJ4

<img width="465" alt="image" src="https://github.com/tonikelope/megabasterd/assets/47494005/14e0ef4f-606b-4ed0-8926-7e3ac00dcf11">

**Comment by KnightChaser** [comment]
@bovirus 

The translation seems to be applied correctly and there is nothing to do more currently. (I'll revise my translation if I or other Korean users discover translation points to be fixed or improved further, but looks okay for now.)

Well, if you are okay, please leave my ID in the About section like other translation contributors later!

**Comment by rktfier** [comment]
Here's the updated files:
[Language_Turkish.txt](https://github.com/tonikelope/megabasterd/files/11714668/Language_Turkish.txt)

**Comment by bovirus** [comment]
@rktfier 

Is it right

         _addTranslation("Korean", "한국인 (KR)");

**Comment by rktfier** [comment]
Here's the updated file:
[Language_Turkish.txt](https://github.com/tonikelope/megabasterd/files/11715741/Language_Turkish.txt)

**Comment by bovirus** [comment]
@rktfier

But this is Tirkist

         _addTranslation("Korean", "**한국어 (KR)**");

**Comment by rktfier** [comment]
> @rktfier
> 
> Please take care that to go Settings and select language you should knwo the language.
> 
> The idea could be translate in native language with abbreviation of the language. Example in Italian
> 
> Cinese (CH) Tedesco (DE) Italiano (IT)
> 
> etc.
> 
> New jar available.

Changes were made based on this.

**Comment by bovirus** [comment]
@rtkfier

Yes sure. The problem is not "(KR)".
Is your intention leave in original Korean language name "한국어" instead Turkish langaige (ex. "Kore")?

**Comment by rktfier** [comment]
Please refer to lines 47, 108, 113, 120, 263, 294 to see how they were made. I'm continuing the trend.

**Comment by bovirus** [comment]
@rktfier 

Ok. Thanks.

**Comment by NotSoMik** [comment]
Has it already been translated into Italian?

**Comment by bovirus** [comment]
@NotSoMik

Megabaster is already translated in italian.
The original language is not completed and this thread has the scope to maintain update langauge files.

To use new version replace original jar file with jar file in first post.

**Comment by FabrieI** [comment]
Devam ettirmene sevindim @rktfier bir süredir takip edemiyordum.Güncelleme sende ise ben müdahele etmiyorum şimdilik.

Teşekkür ettim.

**Comment by rktfier** [comment]
Merhaba, sonunda taşnıştığımıza memnun oldum. Doğrusunu söylemek gerekirse benim de çok yaptığım bir şey yok. Sizin yaptığınız çevirilerde uyuşmazlıklar buldum, fakat bunları düzeltmeye vaktim olmadığından sadece arada bir yeni string'leri çeviriyorum. Buradaki çevirileri ayarlayan arkadaş bana farklı bir proje daha attı. Çevirisinde yer almak isterseniz sizi yönlendirebilirim, hatta üzerinde birlikte çalışabiliriz.

**Comment by FabrieI** [comment]
> Merhaba, sonunda taşnıştığımıza memnun oldum. Doğrusunu söylemek gerekirse benim de çok yaptığım bir şey yok. Sizin yaptığınız çevirilerde uyuşmazlıklar buldum, fakat bunları düzeltmeye vaktim olmadığından sadece arada bir yeni string'leri çeviriyorum. Buradaki çevirileri ayarlayan arkadaş bana farklı bir proje daha attı. Çevirisinde yer almak isterseniz sizi yönlendirebilirim, hatta üzerinde birlikte çalışabiliriz.

Memnuniyet duydum,Hangi proje göz atmak isterim.
İletişim için adresim profilde mevcut istediğin zaman mesaj bırakabilirsin.

**Comment by SEE2097** [comment]
@tonikelope 
Could you please add the Bengali language in your next update? It would be very helpful for Bengali speakers. I hope to hear from you soon.
[Language_Bengali.txt](https://github.com/tonikelope/megabasterd/files/12061198/Language_Bengali.txt)

**Comment by bovirus** [comment]
@SEE2097


There were some errors in language file (use of \ n with a space after \ and befroe n).
Fixed

Please test thish two jar file (please extract from archive and overwrite current .jar file)

**- Add bengali language - Standard fonts**
https://mega.nz/file/DEgQnKRa#XtDzysy-oO3UD-Yr2Pi7di_wqBFugi_8rim8ivR9IDQ

**- Add Bengali language - Alternative fonts**
https://mega.nz/file/vNZn1QqJ#xLfVZayYS6fK1TAVy2JJbfSfZ4NcLP1PlefHtVD2bRE

Pleas report me qwhat is OK about etxt/menu fonts.

Thanks.

**Comment by SEE2097** [comment]
> @SEE2097
> 
> There were some syntax errors in your language file (use of "\ n" with a space after \ and before n). Fixed.
> 
> Please test these two jar files (please extract from archive and overwrite current .jar file)-
> 
> **- Add Bengali language - Standard fonts** https://mega.nz/file/DEgQnKRa#XtDzysy-oO3UD-Yr2Pi7di_wqBFugi_8rim8ivR9IDQ
> 
> **- Add Bengali language - Alternative fonts** https://mega.nz/file/vNZn1QqJ#xLfVZayYS6fK1TAVy2JJbfSfZ4NcLP1PlefHtVD2bRE
> 
> Please report me what of these two file is OK about text/menu fonts.
> 
> Thanks.

Unfortunately, none of these files worked. I could not see any language other than errors. I have marked the places where I can see the no errors in the images. 

The attached images is sequentially placed as Bengali language Standard fonts & Bengali language Alternative fonts.

![Bengali Standard](https://github.com/tonikelope/megabasterd/assets/139624452/0d14982a-e346-4b87-82a0-6971d98db62d)
![Bengali Alternative](https://github.com/tonikelope/megabasterd/assets/139624452/c9d7019b-bc65-4bdd-b2e9-28a99b74b970)

**Comment by bovirus** [comment]
@SEE2097 

Sorry but there is no other option then what I explained (FONT= DEFAULT or ALTERNATIVE).
The translation seems it's ok (see on top elft) but the problem seems the font.
I believe that only tonikelope can check and fix.

**Comment by SEE2097** [comment]
> @SEE2097
> 
> I should a dd new font (ttf file).
> 
> Please try this build
> 
> https://mega.nz/file/ONwgUSAD#AQqERFcHyOEkm0M-dzfHWjA1MoJuzl9V4MQsBCn8rHI
> 
> Note; seems that there is an issue (it happen also with original build 7.74 and chinese language) with two option in tray bar (appear some small boxes).

No luck, brother! It still didn't work. I think some special code is needed for success. As you mentioned, only @tonikelope can handle this. Thank you for your efforts.
the errors are same as before 
![Still Errors](https://github.com/tonikelope/megabasterd/assets/139624452/f67cb54e-b0df-4d46-948f-5f3388a1bbf9)

@tonikelope please help!

**Comment by SEE2097** [comment]
> @SEE2097
> 
> Are you sure to replace java file? It seems an old version of java file
> 
> Please check the screenshot below. Is latest java file that I posted on Mega with Bengali language selected and the screen after Megabasterd restart.
> 
> ![image](https://user-images.githubusercontent.com/1262554/253885202-470f231b-d712-4a0a-8e8e-fe69174b6cf3.png)
> 
> Please note that in your configuration is not set as default Bengali font (for this is not see as expected).


@bovirus 
Oh my bad! Thank you for your efforts again! Yes, almost okay, but the problem is that many letters are breaking up in the font. It would be helpful if you could check my previously submitted "Bengali language txt" file, which will help you understand the problem. Many thanks, brother.

**Comment by bovirus** [comment]
@SEE2097 

Please don't quote previous message.
Is not necessary.
Just reply.

Please add a screenshot to highlight what you mean.

**Comment by SEE2097** [comment]
@bovirus 

i have marked almost all of the wrong broken letters/words in red square boxes. and gave some correct formation of those letters/words in green boxes (couldn't write all the wrong ones lack of space) 

![IMG_20230717_181330](https://github.com/tonikelope/megabasterd/assets/139624452/a4d7a885-d251-442f-967c-1a2220e6bfac)

**Comment by bovirus** [comment]
@SEE2097 

Please provide me a ttf file that could be good for you.

**Comment by bovirus** [comment]
@SEE2097 

The base font file (no bold no italiuc) is "SUTOM___.TTF" or "SutonnyOMJ.ttf"?

**Comment by bovirus** [comment]
@SEE2097 

No good news

![image](https://github.com/tonikelope/megabasterd/assets/1262554/2cfa1dd4-5e5f-4151-afcf-11904e79a6a3)

![image](https://github.com/tonikelope/megabasterd/assets/1262554/6f5f5ad8-a41e-4f50-9ee5-460c215cdee3)

**Comment by SEE2097** [comment]
@bovirus 

The first image looks almost perfect (good to go) but 1/3 of the sentence is cutoff from below.  if you want to use another font then let me know. or you can instruct me how to change the fonts in the application (as i don't have any prior coding knowledge at all! just working as a regular translator). 
try using these please
[Benglai_Language.zip](https://github.com/tonikelope/megabasterd/files/12075284/Benglai_Language.zip)

**Comment by bovirus** [comment]
@SEE2097 

When you post please always explain what font in the package I should use.

**Comment by SEE2097** [comment]
@bovirus 
Which font did you use in the first image? It's excellent! It's working, but it needs some tweaks. I think that English fonts aren't supported, so I am providing a new .txt file without any English words, hope the work is finished after that. Let's cross our fingers!
Integrate the following .txt file with the first image font that you have used. 
[Language_Bengali.txt](https://github.com/tonikelope/megabasterd/files/12078353/Language_Bengali.txt)

**Comment by bovirus** [comment]
@SEE2097 

There were some error in language file "\ n ". "\n " instead "\n" and "" starting strings.

But anyway there some parts of GUi that sue English character (like folder name, values, command in tray bar, and show wrong).

It's important to find the right font that contains the necessary charcater sets.

Language - String issue fixed. 
[Language_Bengali2.txt](https://github.com/tonikelope/megabasterd/files/12078639/Language_Bengali2.txt)

Screenshot

![image](https://github.com/tonikelope/megabasterd/assets/1262554/89db6348-83f5-4635-aa74-d021606728bd)

![image](https://github.com/tonikelope/megabasterd/assets/1262554/0b621b0c-f2e9-49ed-8292-04fc0731918a)

**Comment by SEE2097** [comment]
@bovirus 
This is getting too complicated for me. I'm going to get some rest now.
Grazie mille per il tuo supporto, fratello.

**Comment by linkea131** [comment]
@bovirus @tonikelope

how to Updated, Chinese language

**Comment by bovirus** [comment]
@linkea131 

Get chinese language from archive in first post.
Check if in chiense language there are empty strings to fill or strings to revise.
Post here the new chinese language.

**Comment by linkea131** [comment]
@bovirus @tonikelope

![2E7{F_Y1WDKM3B$4D4EJT5M](https://github.com/tonikelope/megabasterd/assets/50718406/a1043957-887a-4b97-8dca-c5506dbedc71)
![$ $N4OSOLAQ3 RJDLJG8HOM](https://github.com/tonikelope/megabasterd/assets/50718406/ed81af8d-e850-4018-81a6-f1c617ec1dbe)
![YU__7OTYSWT{0`D{IQ117C2](https://github.com/tonikelope/megabasterd/assets/50718406/d8a20a18-9ddb-4fa9-b230-e24bc6bb4084)

I need these characters inside the language text

Because I didn't find him in the language file

**Comment by bovirus** [comment]
@linkea131

There many strings that hardcoded in source file and cannot be translated (are not available in language file) if the author didn't change source code to sue external language file.

@tonikelope 

Pelase fix one tome for all using latest english master (available in first post) and otehr language (always availble in first post).
You current english master is quite old didn't contain all strings.
Please modify the source code to translate as much as possible the current strings untranslated.
Thanks.

**Comment by bovirus** [comment]
Relase new version 7.76
I'm afraid but the author not fixed the issues about language.
Revise source code, language filesd, and build anew jar beta (available in first post).-

**Comment by zYxDevs** [comment]
> 

Add Indonesian Languages:
https://pastebin.com/4kZXtK7x

**Comment by bovirus** [comment]
@zYxDevs

You language file is incomplete (256 strings) because you sued yenglish mastre by author that is not complete.

Please read carefully first post.
Please use the file posted in first post /get language archive and compare english file with your file)..
The master english language is 316 strings.

Whhen you will post language with 316 strings I will add your language.
But pelase take care that I'm not sure that font will be ok (could be the same problem like Bengali).

**Comment by NieckLikesCode** [comment]
Updated and fixed German translation after a long while
[Language_German.txt](https://github.com/tonikelope/megabasterd/files/12387667/Language_German.txt)

**Comment by CyberMSX** [comment]
Portuguese (Brazil) translation:
[Language_Portuguese_Brazilian.txt](https://github.com/tonikelope/megabasterd/files/13167828/Language_Portuguese_Brazilian.txt)

**Comment by Polytelus** [comment]
> Portuguese (Brazil) translation: [Language_Portuguese_Brazilian.txt](https://github.com/tonikelope/megabasterd/files/13167828/Language_Portuguese_Brazilian.txt)

Hi @CyberMSX, it appears your translation is all wrong... You're supposed to fill in the blanks next to the strings, not replace them... Plus the terms such as "password" (which would be "Senha" for all intents and purposes) that you didn't translate.

**Comment by AlMeu** [comment]
@tonikelope 

Here is Romanian translation:
[Language_Romanian.txt](https://github.com/tonikelope/megabasterd/files/13326034/Language_Romanian.txt)

**Comment by bovirus** [comment]
@CyberMSX
@AlMeu

It seems that your file are not right.
The first part of the string should remain in english.
The second part is your tarnslation.

Add to english string also

        _addTranslation("(Portogues (Brasilian)", "");
        _addTranslation("Romanian", "");

with your translation

**Comment by Polytelus** [comment]
Hi bovirus,
To add the language string do I need to add a new line in the end of the
file? Or do I just add next to the other languages in the file itself?

Em seg., 13 de nov. de 2023 12:36, bovirus ***@***.***>
escreveu:

**Comment by bovirus** [comment]
@Polytelus 

Just add them at the end.

About teh tranbslation pelase use tehs ame order of english master  file.
The translation should be like

        _addTranslation(" (Retrying in ", "xxx");

where xxx is your translation.
The first part should remain in english.

**Comment by Polytelus** [comment]
[Language_PortugueseBrazil.txt](https://github.com/tonikelope/megabasterd/files/13339487/Language_PortugueseBrazil.txt)
@tonikelope @bovirus 

Here's the new (and revised) Portuguese (Brazil) translation file.
Hope it works!

**Comment by bovirus** [comment]
@Polytelus

New jar with Portoiguese (Brasilian) language available in first post.
Please downlaod and replace original MegaBaster.jr file.

New Language kit and source fiels changes in first post.

Note: please don't use in teh string text characters like "".
""" are delimiter for tehs tring.
Please use ''.

**Comment by Polytelus** [comment]
@bovirus Hi, I noticed what you meant and removed it. Was it at the "handler" line?
By the way, some strings were missing for localization and I've added them, maybe now it will be 100% localized?
I used the updated english master from the very beginning.
[Language_PortugueseBrazilv2.txt](https://github.com/tonikelope/megabasterd/files/13340704/Language_PortugueseBrazilv2.txt)
V2 uploaded above.

**Comment by bovirus** [comment]
@Polytelus 

Please take care about add new strings of using as tab (not sugegsted) and sue instead spaces.
Where did you get these new strings?

        _addTranslation("Save debug info to file -> ", "Salvar informações de depuração para arquivo -> ");

        _addTranslation("CREATE UPLOAD FOLDER PUBLIC LINK", "CRIAR LINK PÚBLICO PARA ENVIO DE PASTAS");
        _addTranslation("Note: slots consume resources, so use them moderately.", "Nota: Vagas consomem recursos, então as use com moderação.");
        _addTranslation("FORCE SMART PROXY", "FORÇAR SMART PROXY");
        _addTranslation("Note1: enable it in order to mitigate bandwith limit. (Multislot is required)", "Nota 1: Ative para mitigar o limite de banda. (Multislot requerido)");
        _addTranslation("Note2: proxies are selected RANDOMLY.", "Nota 2: Os proxies são selecionados RANDOMICAMENTE.");
        _addTranslation("Forces de use of smart proxy even if we still have direct bandwith avaliable (useful to test proxies).", "Força o uso de smart proxy mesmo que tenhamos banda direta disponível (útil para testar proxies).");
        _addTranslation("COPY ALL DOWNLOAD LINKS", "COPIAR TODOS OS LINKS DE DOWNLOAD");
        _addTranslation("COPY ALL UPLOAD LINKS", "COPIAR TODOS OS LINKS DE ENVIOS");

**Comment by Polytelus** [comment]
By searching my localization file and noticing they seemingly didn't exist
(and because they were still English). Which was weird, so I added them
again. The spaces seem to originate from notepad++, but I have no idea why.

Em seg., 13 de nov. de 2023 17:22, bovirus ***@***.***>
escreveu:

**Comment by bovirus** [comment]
@Polytelus 

Thanks for new strings.

English/Italian/Portoguese Brasilian language files updated
New jar build. Please test it.

**Comment by CyberMSX** [comment]
Thank you for not giving me credit even though you tried to help the translation :(

**Comment by bovirus** [comment]
@CyberMSX 

Please take acre that if soem strings were added in language file are not shos as translated in GUI.,
it depend by instruction used in java code to show it.

@AlMeu

Could you please complete Romanian translation? Thanks.

**Comment by AlMeu** [comment]
@tonikelope & @bovirus

Here is Romanian translation fixed:
[Language_Romanian_Fixed.txt](https://github.com/tonikelope/megabasterd/files/13417827/Language_Romanian_Fixed.txt)

and English version with Romanian added:
[Language_English_RO.txt](https://github.com/tonikelope/megabasterd/files/13417836/Language_English_RO.txt)

I hope now it is OK.

**Comment by linkea131** [comment]
@bovirus @tonikelope

Updated, Chinese language (Some fixes,dialog December 08 2023)

https://github.com/linkea131/megabasterd_translate

**Comment by linkea131** [comment]
@bovirus @tonikelope
There are new language texts to be added
![QQ图片20231216201601](https://github.com/tonikelope/megabasterd/assets/50718406/ca7d13cc-2f24-4166-99f9-b182615546b6)
![QQ图片20231216201617](https://github.com/tonikelope/megabasterd/assets/50718406/05882910-a976-4c0f-ac59-9af0e1c4d53c)
![QQ图片20231216201622](https://github.com/tonikelope/megabasterd/assets/50718406/9244b9a9-d64a-4226-a5fe-daa8bcf8eaba)
![QQ图片20231216201625](https://github.com/tonikelope/megabasterd/assets/50718406/61a65391-c862-47ee-8203-aa406e81184c)
![QQ图片20231216201629](https://github.com/tonikelope/megabasterd/assets/50718406/fb9bb08d-7e96-4975-899f-4d8b2ea2a2c2)
![QQ图片20231216201634](https://github.com/tonikelope/megabasterd/assets/50718406/0041c8ec-001d-4b2e-9f47-e3f346ee9863)
![QQ图片20231216201640](https://github.com/tonikelope/megabasterd/assets/50718406/7eaf0f65-4961-4ad5-8741-673c1943ce50)
![QQ图片20231216201644](https://github.com/tonikelope/megabasterd/assets/50718406/c670b768-9eaa-485b-9203-541ea782c650)
![QQ图片20231216201647](https://github.com/tonikelope/megabasterd/assets/50718406/344e2890-113a-4f8e-a225-689f611309c1)
![QQ图片20231216201651](https://github.com/tonikelope/megabasterd/assets/50718406/f6100cdb-04ec-468e-9ed2-619a6c913cbc)
![QQ图片20231216201654](https://github.com/tonikelope/megabasterd/assets/50718406/6c92f194-fc63-4f74-921c-112d3f759cf2)
![QQ图片20231216201701](https://github.com/tonikelope/megabasterd/assets/50718406/7079a731-c9f0-4934-9891-cb455e0a626b)
![QQ图片20231216201705](https://github.com/tonikelope/megabasterd/assets/50718406/5b11913d-0f58-4821-88a1-3a90bdd0fad7)
![QQ图片20231216201709](https://github.com/tonikelope/megabasterd/assets/50718406/c01356e5-ce9f-49e3-b734-77ebc1240ce4)
![QQ图片20231216201711](https://github.com/tonikelope/megabasterd/assets/50718406/a97ff185-e524-4d15-b675-0fea8019e08e)
![QQ图片20231216201716](https://github.com/tonikelope/megabasterd/assets/50718406/8acadb7a-bf5a-47a3-8804-eb9d8cc1cde4)
![QQ图片20231216201721](https://github.com/tonikelope/megabasterd/assets/50718406/c77981b1-1c0e-4674-9337-149278d301ed)
![QQ图片20231216201724](https://github.com/tonikelope/megabasterd/assets/50718406/230f9735-ac99-49bd-b178-af467d0ec120)
![QQ图片20231216201727](https://github.com/tonikelope/megabasterd/assets/50718406/7c01ace0-3d80-40ee-9094-1d29ac8f496b)
![QQ图片20231216201730](https://github.com/tonikelope/megabasterd/assets/50718406/8fe770d4-84b4-4699-9ee1-bf273f42c6f7)
![QQ图片20231216201735](https://github.com/tonikelope/megabasterd/assets/50718406/6d481485-2c1d-4a5c-bdef-5cf45abc0b45)
![QQ图片20231216201740](https://github.com/tonikelope/megabasterd/assets/50718406/5d40c0f5-76ca-4521-8509-1b0c063b63cc)
![QQ图片20231216201744](https://github.com/tonikelope/megabasterd/assets/50718406/8d529a39-8ce4-4fdf-9b5e-9035e2b6c901)
![QQ图片20231216201747](https://github.com/tonikelope/megabasterd/assets/50718406/2a6c339b-0238-474b-8200-fd4894935020)

**Comment by bovirus** [comment]
@linkea131

Some of the string are related to language updates that are not applied by the project author.
Some strings are nerw

It's quite difficult to follow the author if they didn't ordfganize tehd evelopment about language issues.

The current original language in LabelTranslation file 

- are not updated
- are nor organized by alphabetical order
- etc,.

I will study teh changes and I will prpose new master english file to update all other files.

@tonikelope 
Please follow our language develòopment.
We are interested to maintain language updatee.
But if you didn't apply for us it's a long and boring job.

**Comment by bovirus** [comment]
@linkea131 

I updated English master and Italian language.
I compield new JAR but it seems that strings also are iun language file are not translated.

Please check screenshot.

![image](https://github.com/tonikelope/megabasterd/assets/1262554/5795aac9-159e-4cfe-81ba-d6927661a030)

@tonikelope 

Please check source and the reason why strings are not translated.

**Comment by tonikelope** [comment]
**Heads-up to all translators — MegaBasterd's i18n system has been overhauled.**

### What changed

The translation table was previously a ~2,140-line block of hardcoded
`_addTranslation("English text", "Localized text")` Java calls inside
`LabelTranslatorSingleton.java`, one private method per language. That
made every translation update require editing Java source and rebuilding
the project.

It now lives in plain UTF-8 `.properties` files under `src/main/resources/i18n/`:

```
messages.properties      <- English canonical (also the key index)
messages_es.properties   <- Spanish
messages_it.properties   <- Italian
messages_de.properties   <- German
messages_hu.properties   <- Hungarian
messages_tr.properties   <- Turkish
messages_zh.properties   <- Chinese
messages_vi.properties   <- Vietnamese
```

Internally it's a Java `ResourceBundle` with a custom UTF-8 `Control`
(default Java 8 behavior is ISO-8859-1, hence the explicit override).
At runtime the active locale's bundle is consulted first, with English
as the fallback for any missing key.

`LabelTranslatorSingleton.translate(String)` still works exactly as
before — it now just inverts the English bundle at boot and delegates
to the new API. **No call site or `.form` file had to change** for
existing UI to keep translating.

### How to contribute a translation now

1. Fork the repo and open `src/main/resources/i18n/messages_<your_lang>.properties`.
2. Each entry has the format `key=Your translation`, with a `# EN: ...`
   comment above showing the canonical English text for context.
3. Edit the value to the right of `=`. **You don't need to write Java
   anymore, and you don't need to recompile** — just save the file and
   open a PR.
4. `\n` becomes a real newline. `{0}`, `{1}`, ... are runtime
   placeholders (used in dynamic messages like
   `ui.dynamic.restart_countdown=Restart ({0} secs...)`) — keep them
   verbatim.
5. If your language isn't in the list yet, copy `messages.properties`
   to `messages_<iso_code>.properties` (e.g. `messages_pt.properties`
   for Portuguese) and translate. Then ping in this thread so we add
   the entry to the Settings combo.

### About the AI-assisted seeds

About 100 new keys were added in this refactor — entries that
previously rendered in English in every locale because nobody had
added them to the per-language tables (`COPY ALL DOWNLOAD LINKS`,
`COPY ALL UPLOAD LINKS`, `FOLDER CACHE` prompts, master password
setup, proxy tooltips, file chooser titles, "Export accounts"
flow, retry messages, dark mode, 2FA dialog title, etc.).

Spanish is fully translated. **German, Italian, Hungarian, Turkish,
Chinese and Vietnamese were seeded with AI-assisted translations
and need a human review pass.** If you maintain one of those
locales, please grep your bundle for the `# Seeded ... 8.37 i18n
refactor` divider — every entry below it is fair game for
correction.

### Tooling included

Two verifier scripts live under `scripts/` for anyone who wants to
sanity-check their work:

* `scripts/verify_i18n_refs.py` — every `I18n.tr("key")` and
  `translate("English literal")` call in the Java source must
  resolve to a bundle entry. Currently 0 misses.
* `scripts/verify_i18n_forms.py` — every `text` / `title` /
  `toolTipText` property in the NetBeans `.form` files must
  resolve. Currently 0 misses.

If you submit a translation PR, running these locally takes a few
seconds and catches typos / missing keys.

Thanks for keeping MegaBasterd multilingual!

**Comment by tonikelope** [comment]
**v8.40 i18n changelog** — 9 new keys + 4 updated literals across the SmartProxy and Quota Recovery dialogs.

All 8 bundles (`messages.properties`, `messages_de`, `messages_es`, `messages_hu`, `messages_it`, `messages_tr`, `messages_vi`, `messages_zh`) ship with what I believe are reasonable translations, but native speakers please review — particularly the cross-referenced tooltips, which need to read naturally to deliver the disambiguation.

### New keys

- `settings.smartproxy.refresh.tooltip` — long-form tooltip for the `Settings -> SmartProxy` refresh-time spinner. Must explicitly mention it is NOT the same as the post-509 SmartProxy window in the Quota recovery dialog (Edit menu).
- `ui.quota.batch_size_label` — short label for a new spinner: `Concurrent probes:`.
- `ui.quota.batch_size.tooltip` — long-form tooltip: how many TCP connect probes run in parallel, range 1..100, default 20.
- `ui.quota.test.cancelled` — `[INFO] Probe cancelled.` line emitted to the test output area when a probe is interrupted (e.g. the dialog closes mid-test).
- `ui.quota.save_working_button` — new button label: `Save working proxies to custom list`.
- `ui.quota.save_working.tooltip` — explains the button: rewrite inline `IP:PORT` entries to keep only those that just passed the test, while preserving `#URL` source lines.
- `ui.quota.save_working.title` — confirm dialog title.
- `ui.quota.save_working.confirm` — confirm dialog body, takes `{0}` = number of working proxies.
- `ui.quota.save_working.done` — `[INFO] Saved {0} working proxies to the custom list.` output line.
- `ui.quota.save_working.error` — `[ERROR] Could not save working proxies: {0}` output line.

### Updated literals (existing keys, new values)

- `proxy_list_refresh` — was `Proxy list refresh (minutes):`, now `Re-download proxy list every (minutes):`. The label was renamed for clarity (it kept being confused with the post-509 window in the Quota recovery dialog). The English literal change had to match the `LabelTranslatorSingleton` English-literal -> key lookup, so `SettingsDialog.java`'s `jLabel8.setText` and `SettingsDialog.form`'s `text` Property were updated in lockstep.
- `ui.quota.recheck_window_label` — was `SmartProxy 509 recheck window (s):`, now `Keep SmartProxy active after 509 for (s):`. Same disambiguation reason.
- `ui.quota.recheck_window.tooltip` — body rewritten to mention `Settings -> SmartProxy` explicitly so the cross-reference is symmetric.
- `ui.quota.test.testing_count` — added a `{1}` placeholder for the new concurrent-probes count. Was `Testing {0} proxy entries (3 s TCP connect timeout each)...`, now `Testing {0} proxy entries with {1} concurrent probes (3 s TCP connect timeout each)...`.

Release notes: https://github.com/tonikelope/megabasterd/releases/tag/v8.40

Thanks as always to the translators here. Open to PRs / suggestions on any of the new strings.

**Comment by tonikelope** [comment]
New i18n keys added in master (aa796ed) as part of the #757 fixes. EN root + ES/IT localised, the rest carry the EN string as a placeholder pending a translation pass.

**New keys**
- `settings.smartproxy.ban_time.tooltip` — explains 0/1..9/10..3600 ban-time bands on the SmartProxy spinner
- `settings.smartproxy.force.tooltip` — PASSIVE vs FORCE mode tooltip on the FORCE SMART PROXY checkbox

**Updated key (clarified wording)**
- `ui.quota.batch_size.tooltip` — now explicitly states the spinner only affects the proxy Test action (not runtime SmartProxy concurrency) and that the value is persisted across sessions

**Bundles needing translation**: `messages_de.properties`, `messages_hu.properties`, `messages_tr.properties`, `messages_vi.properties`, `messages_zh.properties` (all three keys); `messages_de.properties`, `messages_hu.properties`, `messages_tr.properties`, `messages_vi.properties`, `messages_zh.properties` also have an EN-placeholder for the updated `ui.quota.batch_size.tooltip`.

**Comment by tonikelope** [comment]
v8.44 ships three new keys (#758 fixes):

- `settings.smartproxy.timeout.tooltip` -- explains the `[3, 120]` band on the Proxy-timeout spinner.
- `ui.smartproxy.refresh_now` -- "Refresh proxy list now" button on the SmartProxy tab.
- `ui.smartproxy.refresh_now.tooltip` -- explains the button is a manual override of the auto-refresh interval and that the fetch runs in the background.

EN/ES/IT localised. DE/HU/TR/VI/ZH carry best-effort translations -- refinements via this thread are welcome as usual.

**Comment by tonikelope** [comment]
Heads up to translators: [v8.50](https://github.com/tonikelope/megabasterd/releases/tag/v8.50) (fixes #770) adds three new keys to every `messages*.properties` bundle:

```
always_reload_mega_folders=Always reload MEGA folders instead of using cached folder data
default_download_directory=Default download directory
java_properties_utf_8_filter=Java properties (UTF-8) (*.properties)
```

EN / ES / IT are hand-authored. DE / HU / TR / VI / ZH were translated in the same style as the previous baselines from this thread — if you spot something off in your locale, a PR or a paste in this thread is welcome.

8.50 also fills in proper translations for the existing `user` / `api_key` keys, which were placeholder English literals (`"User"`, `"API-KEY"`) in `messages_es`, `messages_de`, `messages_vi`, and (for `api_key` only) `messages_zh`. The new ELC accounts-table header routing now actually surfaces localised text there. Values landed: ES `Usuario`/`Clave API`, DE `Benutzer`/`API-Schlüssel`, VI `Người dùng`/`Khóa API`, ZH `API 密钥`. IT, HU, TR were already complete.

Also in 8.50: `LabelTranslatorSingleton` now HTML-wraps the SmartProxy WARNING JLabel before assigning it (literal `\n` is converted to `<br>`), so you can break the long *WARNING: Using proxies or VPN...* paragraph (`warning_using_proxies_or_vpn_to_bypass_mega_s_daily`) into multiple lines using `\n` in your locale's value — the Italian bundle has been doing this since v8.46, the change makes that actually render. No new key, just a rendering fix at the call site.

The ELC accounts table column headers (`Host` / `User` / `API-KEY`) are also now localised — they reuse the existing `host` / `user` / `api_key` keys, so no further translator action needed there.

(v8.49 was rolled into v8.50; that tag/release no longer exists.)

**Comment by tonikelope** [comment]
v8.54 adds 2 new keys (post-finish command checkboxes in Settings &rarr; Advanced):

- `execute_this_command_when_all_downloads_finish`
- `execute_this_command_when_all_uploads_finish`

EN and ES are translated; IT, DE, HU, TR, VI and ZH currently carry the English placeholder pending translator review. Thanks for keeping these bundles in sync!
